app/models/concerns/reviewable_action_builder.rb

```diff
@@ -241,6 +241,101 @@ def perform_convert_to_pm(performed_by, _args)
     create_result(:success, :rejected, [created_by_id], false)
   end
 
+  # Calculate the final reviewable status based on the action logs. This method
+  # assumes that all_bundles_actioned? has already returned true.
+  #
+  # @return [Symbol] The calculated final status (:ignored, :rejected, :approved, or :pending)
+  def calculate_final_status_from_logs
+    statuses = reviewable_action_logs.pluck(:status).uniq
+
+    return :pending if statuses.empty?
+
+    ignored_value = Reviewable.statuses["ignored"]
+    rejected_value = Reviewable.statuses["rejected"]
+    approved_value = Reviewable.statuses["approved"]
+
+    return :ignored if statuses.all? { |s| s == "ignored" || s == ignored_value }
```

Comment:
Do we need that string comparison? It is new model so I assume we will have consistent data
```ruby
 return :ignored if statuses.all? { |status| status ==  Reviewable.statuses["ignored"]}
```